Backpack/Nudger/Classes/BPKNudger.m

```diff
@@ -0,0 +1,297 @@
+
+
+    [self addSubview:self.stackView];
+    self.stackView.translatesAutoresizingMaskIntoConstraints = NO;
+
+    // Reserve space for 2 digits. Which two digits does not matter
+    // because we use tabular(i.e. monospaced) digits.
+    CGFloat minWidth = [self.label widthForLabel:@"10"];
+    [NSLayoutConstraint activateConstraints:@[
+        [self.topAnchor constraintEqualToAnchor:self.stackView.topAnchor],
+        [self.leadingAnchor constraintEqualToAnchor:self.stackView.leadingAnchor],
+        [self.stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
+        [self.stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+        [self.label.widthAnchor constraintGreaterThanOrEqualToConstant:minWidth],
+    ]];
+    self.accessibilityTraits = UIAccessibilityTraitAdjustable;
+    self.isAccessibilityElement = YES;
+
+    [self updateDisplay];
+}
+
+#pragma mark - Properties
+
+- (UIStackView *)stackView {
+    BPKAssertMainThread();
+    if (_stackView) {
+        return _stackView;
+    }
+
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.alignment = UIStackViewAlignmentCenter;
+    stackView.spacing = BPKSpacingMd;
+
+    [stackView addArrangedSubview:self.decrementButton];
+    [stackView addArrangedSubview:self.label];
+    [stackView addArrangedSubview:self.incrementButton];
+
+    _stackView = stackView;
+
+    return _stackView;
+}
+
+- (BPKButton *)decrementButton {
+    BPKAssertMainThread();
+    if (_decrementButton) {
+        return _decrementButton;
+    }
+
+    BPKButton *decrementButton = [[BPKButton alloc] initWithSize:BPKButtonSizeDefault style:BPKButtonStyleSecondary];
+    [decrementButton setImage:[BPKIcon smallTemplateIconNamed:BPKSmallIconNameMinus]];
+
+    _decrementButton = decrementButton;
+    _decrementButton.accessibilityLabel = self.configuration.decrementButtonLabel;
+    [_decrementButton addTarget:self action:@selector(didTapDecrement:) forControlEvents:UIControlEventTouchUpInside];
+
+    return _decrementButton;
+}
+
+- (BPKNudgerLabel *)label {
+    BPKAssertMainThread();
+    if (_label) {
+        return _label;
+    }
+
+    BPKNudgerLabel *label = [[BPKNudgerLabel alloc] initWithFrame:CGRectZero];
+    label.text = @"0";
+
+    label.frameProvider = self;
+    _label = label;
+
+    return _label;
+}
+
+- (BPKButton *)incrementButton {
+    BPKAssertMainThread();
+    if (_incrementButton) {
+        return _incrementButton;
+    }
+
+    BPKButton *incrementButton = [[BPKButton alloc] initWithSize:BPKButtonSizeDefault style:BPKButtonStyleSecondary];
+    [incrementButton setImage:[BPKIcon smallTemplateIconNamed:BPKSmallIconNamePlus]];
+
+    _incrementButton = incrementButton;
+    _incrementButton.accessibilityLabel = self.configuration.incrementButtonLabel;
+    [_incrementButton addTarget:self action:@selector(didTapIncrement:) forControlEvents:UIControlEventTouchUpInside];
```

Comment:
Could we avoid duplicating some of this button logic between the two buttons?
Maybe it would do more harm (in terms of readability) than good (in terms of reusing code)

Reply:
YAGNI